code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -162,11 +189,13 @@
 				Goto(target,move_to_delay,minimum_distance)//Otherwise, get to our minimum distance so we chase them
 		else
 			Goto(target,move_to_delay,minimum_distance)
-		if(isturf(loc) && target.Adjacent(src))	//If they're next to us, attack
-			AttackingTarget()
-		return 1
+		if(target)

```

Comment:
I don't know, it's not necessary as it's checked above, probably a hold over from when I was debugging the AI.